example/src/main/java/ably/io/tutorial/ExampleActivity.java

```diff
@@ -34,12 +39,28 @@ protected void onCreate(Bundle savedInstanceState) {
         } catch (AblyException e) {
             e.printStackTrace();
         }
+        findViewById(R.id.btPublish).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                /* Get message content from EditText */
+                String message = messageEditText.getEditableText().toString();
+                /* Check if message is not empty */
+                if (!TextUtils.isEmpty(message)) {
+                    try {
+                        Log.v(getClass().getCanonicalName(), "Publishing message...");
```

Comment:
Yes, but I assumed the developer who is running those samples is now in the role of use, so he will see these messages. If you want, the simplest fix will be to show a Toast message.